python/observation-publisher/publisher/prediction_log_consumer.py

```diff
@@ -34,22 +39,68 @@ def close(self):
         raise NotImplementedError
 
     def start_polling(
-        self, observation_sink: ObservationSink, inference_schema: InferenceSchema
+        self,
+        observation_sinks: List[ObservationSink],
+        inference_schema: InferenceSchema,
     ):
         try:
+            buffered_logs = []
+            buffered_max_duration_seconds = 60
+            buffer_start_time = datetime.now()
             while True:
                 logs = self.poll_new_logs()
                 if len(logs) == 0:
                     continue
-                df = log_batch_to_dataframe(logs, inference_schema)
-                observation_sink.write(df)
+                buffered_logs.extend(logs)
+                buffered_duration = (datetime.now() - buffer_start_time).seconds
+                if (
+                    len(buffered_logs) < self.buffer_capacity
+                    and buffered_duration < buffered_max_duration_seconds
+                ):
+                    continue
+                df = log_batch_to_dataframe(buffered_logs, inference_schema)
+                most_recent_prediction_timestamp = df[
+                    PREDICTION_LOG_TIMESTAMP_COLUMN
+                ].max()
+                MetricWriter().update_last_processed_timestamp(
```

Comment:
what is the purpose of updating this?

Reply:
This is mainly to track whether the consumer is lagging. If we see that the last processed timestamp is far earlier than the current time, then it's an indicator that the processing speed is not fast enough.